### ISSUE TITLE
Mem leaks observed with LSAN 

### ISSUE THREAD

**Issue by PaoloBenigni** [liyunfan1223/mod-playerbots#719]
I finally managed to get some clear measuements of the worldserver checking for memory leaks using lsan
See output [here](https://gist.github.com/PaoloBenigni/52983ab4882ced3865806110245eaa75).

**Commit hash**
The hash of the current commit.

**To Reproduce**
1. compile with the following 
```
cmake ../ \
	-DCMAKE_INSTALL_PREFIX=$HOME/source_core/env/dist/ \
	-DCMAKE_C_COMPILER=/usr/bin/clang \
	-DCMAKE_CXX_COMPILER=/usr/bin/clang++ \
	-DWITH_WARNINGS=1 \
	-DTOOLS_BUILD=none \
	-DSCRIPTS=static \
	-DMODULES=static \
	-DCMAKE_CXX_FLAGS="-fsanitize=leak -fno-omit-frame-pointer"\
	-DCMAKE_EXE_LINKER_FLAGS="-fsanitize=leak"\
	-DCMAKE_MODULE_LINKER_FLASGS="-fsanitize=leak" &&
```
2. Run the program and capture the output. 

**Desktop (please complete the following information):**
-Ubuntu WSL on windows server.

**Comment by davvvve1** [comment]
Im not expert, but if we could change AiPlayerbot.BotActiveAlone = 10  to zero (now is 10 minimum) wouldnt it be easier to find where 
the memory leak is ?  I ment if it when bots using lfg/bg or running around like chickens in the world . Just my 2 cent

**Comment by PaoloBenigni** [comment]
Here is the conf I used. 
[playerbots-LSAN-test1.conf.txt](https://github.com/user-attachments/files/17849929/playerbots-LSAN-test1.conf.txt)

Relevant section
```
AiPlayerbot.MinRandomBots = 1000
AiPlayerbot.MaxRandomBots = 1000

AiPlayerbot.BotActiveAlone = 100
AiPlayerbot.botActiveAloneSmartScale = 0
AiPlayerbot.botActiveAloneSmartScaleWhenMinLevel = 1
AiPlayerbot.botActiveAloneSmartScaleWhenMaxLevel = 80

```

**Comment by noisiver** [comment]
> Here is the conf I used. [playerbots-LSAN-test1.conf.txt](https://github.com/user-attachments/files/17849929/playerbots-LSAN-test1.conf.txt)
> 
> Relevant section
> 
> ```
> AiPlayerbot.MinRandomBots = 1000
> AiPlayerbot.MaxRandomBots = 1000
> 
> 
> AiPlayerbot.RandomBotJoinLfg = 1
> 
> AiPlayerbot.RandomBotAutoJoinBG = 0
> 
> AiPlayerbot.BotActiveAlone = 100
> AiPlayerbot.botActiveAloneSmartScale = 0
> AiPlayerbot.botActiveAloneSmartScaleWhenMinLevel = 1
> AiPlayerbot.botActiveAloneSmartScaleWhenMaxLevel = 80
> ```
> 
> No battleground, but lfg is on. I dont think they team up in LFG if a player doesnt, and this had no players logged in at all.

They never do any PvE content without a player. I mean dungeons and raids. Having them do battlegrounds on their own still works, though it's limited to WSG.

**Comment by davvvve1** [comment]
I understand that , but when im not online on the server it takes longer time for the memory to raise , but when i do raids/dungeons its faster . But you probaly know more about that stuff.   I noticed that my client crashes more freqvently . Now i have removed all patches 
on the client . I only have the m patch for the map left and use opengl and its stable now . Gonna try some more . Running Debian 12 as server and Arch linux on my gaming computer. 
AiPlayerbot.MinRandomBots = 3000
AiPlayerbot.MaxRandomBots = 3000


AiPlayerbot.RandomBotJoinLfg = 1

AiPlayerbot.RandomBotAutoJoinBG = 0

AiPlayerbot.BotActiveAlone = 10
AiPlayerbot.botActiveAloneSmartScale = 0
AiPlayerbot.botActiveAloneSmartScaleWhenMinLevel = 1
AiPlayerbot.botActiveAloneSmartScaleWhenMaxLevel = 80
AiPlayerbot.ProbTeleToBankers = 1

# Maps where bots can be teleported to
#AiPlayerbot.RandomBotMaps = 0

with 3 k bots i have 15 ms   .  I only wants bots when i invite them and i want it t0 be simple without any addons .  
Im gonna run this a couple of days to see if its steady . I noticed it isnt much teleporting .

**Comment by davvvve1** [comment]
After 2 hours it still the same ms , and memory is the same .  For my use its great , and the cities is more alive .  When im doing quest  is better cause i dont have to fight for the npc to respawn ;)

**Comment by PaoloBenigni** [comment]
The magnitude of the impact will be user specific. Right now here we need someone with the ability to make sense of the LSAN output.

**Comment by davvvve1** [comment]
> The magnitude of the impact will be user specific. Right now here we need someone with the ability to make sense of the LSAN output.

I totally agree with you . But for now if you need help with a bot you only invite them to a party.  I hoping to find any friends who would play on my server that i can test it some more with other players .

**Comment by davvvve1** [comment]
Uptime 6 Hours 44 minutes 
3 k bots 
Mean 17 ms 
Median 12 ms

**Comment by noisiver** [comment]
Update diff isn't interesting when we're talking about memory leaks, memory usage is.

**Comment by davvvve1** [comment]
> Update diff isn't interesting when we're talking about memory leaks, memory usage is.

Ok ,  but when im gaming that when update diff start rasing , soon after the memory shooting thru the roof . Im new at this. 
My server has been on with 3 k bots and using about 10 gb ram after 11 hour .

**Comment by hermensbas** [comment]
> section

10 isnt minimal bot some bots are forced to be active

**Comment by PaoloBenigni** [comment]
Let me re-run this and Ill report back.

**Comment by PaoloBenigni** [comment]
I cant reproduce this currently anymore. Closing.
